lib/pwa.js

```diff
@@ -251,37 +251,63 @@ exports.findByEncodedStartUrl = function(encodedStartUrl) {
 };
 
 /**
- * Validates and saves the PWA
+ * Creates or Updates a PWA.
+ *
+ * Steps:
+ *  1) Validate Pwa
+ *  2) Update Pwa's Manifest
+ *  3) Save (to get the DB id for following steps)
+ *  4) Update PWA's MetadataDescription
+ *  5) Update PWA's Icon
+ *  6) Save
+ *  7) (in background) Update Lighthouse, Save & Send notifications for new PWAs.
  *
  * @param {Pwa} pwa to update
  * @return {Pwa} the updated PWA
  */
-exports.save = function(pwa) {
-  if (!pwa) {
-    return Promise.reject('pwa is null');
-  }
+exports.createOrUpdatePwa = function(pwa) {
+  const steps = [
+    _ => this.validatePwa(pwa),
```

Comment:
I still need the Promise return functions, I am not sure how it could be rewrite.